### DISCUSSION TITLE
How come rpc eth_getBalance call  sends wrong data about address balance?

### DISCUSSION THREAD

**Discussion by Valentin529** [WalletConnect/walletconnect-monorepo#3036] [Developer Support]
I connected this address  "0xD17d152d9f32bb1Cf33daffc9d67821AbBd612EA" to my web3 project. it's balance in BNB chain 56 is 0.0756. I have sent 40 calls to your rpc using eth_getBalance method and 38 times received correct balance and 2 times receives wrong. 
I attach the screenshots of a call request payload and responce
Call the method with params ->  ![Screenshot 2023-07-07 at 12 27 17](https://github.com/WalletConnect/walletconnect-monorepo/assets/68601123/08d42076-2f37-4f46-bcf8-09f39f26b6c8)

Get response -> ![Screenshot 2023-07-07 at 12 27 27](https://github.com/WalletConnect/walletconnect-monorepo/assets/68601123/195b01b7-9457-4c3b-b086-0fdac55927ac)

Parsed balance  -> ![Screenshot 2023-07-07 at 12 27 36](https://github.com/WalletConnect/walletconnect-monorepo/assets/68601123/0024ba8c-e8e4-4d24-a9fc-05994756796b)

Some times it returns bigger balance, some times it returns smaller. I tried both build in @wagmi/core chains and a custom added chains.  I don't have any issues with testnet though...
I am 100% sure that it is not a parsing issue from my side as I double checked and the response from RPC is not correct. If I catch this situation and send another getbalance request rigth away - it 99.99% of the time returns correct balance.
I ask for your help as it currently it really destroys the UI for users :(

Here are the example of correct balances 
Same request sent -> ![Screenshot 2023-07-07 at 13 41 31](https://github.com/WalletConnect/walletconnect-monorepo/assets/68601123/80e37aeb-f75b-48aa-9145-99716bbbb3a7)
Another response ->  ![Screenshot 2023-07-07 at 13 41 37](https://github.com/WalletConnect/walletconnect-monorepo/assets/68601123/dde7ed75-f314-4fea-916b-2b75f189a50a)

Correct balance -> ![Screenshot 2023-07-07 at 13 41 43](https://github.com/WalletConnect/walletconnect-monorepo/assets/68601123/3dbd871a-84c5-4139-a92f-8f0ee04c7146)

**Comment by crypblizz8** [comment]
Can you provide your W3M setup + how you have called eth_getBalance?

**Reply by Valentin529** [reply]
This is the setup:
`

    import { EthereumClient, w3mConnectors, w3mProvider } from "@web3modal/ethereum";
    import { Web3Modal } from "@web3modal/html";
    import {
      configureChains,
      createConfig,
      readContracts,
      prepareWriteContract,
      writeContract,
      switchNetwork,
      waitForTransaction,
    } from "@wagmi/core";
    import { bsc, bscTestnet  } from "@wagmi/core/chains";
     this.chains =  [bsc]
     this.defaultChains =  [bsc]
     const { publicClient } = configureChains(
        this.chains,
        [w3mProvider({ projectId })],
        { defaultChains: this.defaultChains }
      );
     const wagmiClient = createConfig({
      autoConnect: true,
      connectors: w3mConnectors({ projectId, version: 1, chains: this.chains }),
      publicClient,
    });
    this.ethereumClient = new EthereumClient(wagmiClient, this.chains);
    this.web3modal = new Web3Modal({ projectId }, this.ethereumClient);
    "
    
    This is how i fetch BNB balance: 
    " this.ethereumClient.fetchBalance({ address: "0xD17d152d9f32bb1Cf33daffc9d67821AbBd612EA"})"
    
    npm versions: 
    "@wagmi/core": "^1.3.1",
    "@web3modal/ethereum": "^2.6.0",
    "@web3modal/html": "^2.6.0",
    "viem": "^1.1.8",
    Also tried on versions but issue persists: 
    "@wagmi/core": "^1.3.6",
    "@web3modal/ethereum": "^2.6.2",
    "@web3modal/html": "^2.6.2",
    "viem": "^1.2.9" `

**Reply by Valentin529** [reply]
Any updates?

**Reply by Valentin529** [reply]
@crypblizz8 any updates?

**Comment by Valentin529** [comment]
@crypblizz8  Any updates on this issue?

**Comment by SergeyVolynkin** [comment]
Observed this weird(and scary) issue once on Testnet(Goerli) but it no longer reproduces 🤔
